otcextensions/sdk/dds/v3/datastore.py

```diff
@@ -0,0 +1,26 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+from openstack import resource
+
+
+class Datastore(resource.Resource):
+    base_path = '/datastores/%(datastore_name)s/versions'
```

Comment:
this is not going to provide nice interface. What if we do following:
- datastore type has all props that are passed in the datastore struct to create (`type=DDS-Community`, `version=xxx`, `storage_engine=wiredTiger`)
- datastore.list is overloaded, invokes /versions and yields filled entry for every entry
This way we would be able to provide same/similar interface to RDS